intelij-plugin/src/main/java/tech/pantheon/yanginator/plugin/external/ExternalRules.java

```diff
@@ -0,0 +1,88 @@
+     */
+
+    public static boolean anyOrder(PsiBuilder psiBuilder, int level, Parser... listOfItems) {
+        if (!recursion_guard_(psiBuilder, level, "rule")) return false;
+        boolean parserResultBoolean;
+        //create new array of booleans with length of items that we want to match
+        boolean[] listOfBooleanResults = new boolean[listOfItems.length];
+        //create list out of listOfItems array, so we can remove items that were already matched
+        List<Parser> listOfItemsToParse = new ArrayList<>(List.of(listOfItems));
+        Set<Parser> optionalParsed = new HashSet<>();
+        //enter the section to match
+        PsiBuilder.Marker marker = enter_section_(psiBuilder);
+
+        for (int i = 0; i < listOfItems.length; i++) {
+            parserResultBoolean = false;
+            for (int j = 0; j < listOfItemsToParse.size(); j++) {
+                int oldOffset = psiBuilder.getCurrentOffset();
+                //try parsing item
+                parserResultBoolean = listOfItemsToParse.get(j).parse(psiBuilder, level + 1);
+                if (i == j && parserResultBoolean) {
+                    listOfBooleanResults[i] = true;
+                }
+                int newOffset = psiBuilder.getCurrentOffset();
+                //optional or not optional
+                if (parserResultBoolean && (oldOffset == newOffset)) {
+                    optionalParsed.add(listOfItems[j]);
+                    //try parsing optional
+                    for (Parser item : optionalParsed) {
+                        oldOffset = psiBuilder.getCurrentOffset();
+                        parserResultBoolean = item.parse(psiBuilder, level + 1);
+                        if (parserResultBoolean && oldOffset != newOffset) {
+                            //remove optional item if it was parsed
+                            optionalParsed.remove(item);
+                            listOfBooleanResults[i] = true;
+                            //no need to loop futher, the item has been parsed
+                            break;
+                        } else {
+                            parserResultBoolean = false;
+                        }
+                    }
+                }
+                if (parserResultBoolean) {
+                    listOfBooleanResults[j] = true;
+                    //no need to loop futher, the item has been parsed
+                    break;
+                }
+            }
+            //move to next element and try matching it
+            level += 1;
+        }
+        exit_section_(psiBuilder, marker, null, areAllTrue(listOfBooleanResults));
+        return areAllTrue(listOfBooleanResults);
+    }
+
+    private static boolean areAllTrue(boolean[] listOfBooleanResults) {
+        //simple method to find if all elements are true
+        for (boolean booleanList : listOfBooleanResults) if (!booleanList) return false;
```

Comment:
I wouldn use oneliner here, as there is more than meets the eye...

for (boolean individualResult : individualResults) {
 if (!individualResult) return false;
}